src/streaming/metrics/controllers/MetricsCollectionController.js

```diff
@@ -91,16 +91,21 @@ function MetricsCollectionController(config) {
     }
 
     function setup() {
-
-
         eventBus.on(Events.MANIFEST_UPDATED, update);
-        eventBus.on(Events.STREAM_TEARDOWN_COMPLETE, reset);
+        eventBus.on(Events.STREAM_TEARDOWN_COMPLETE, resetMetricsControllers);
+    }
+
+    function reset() {
+        eventBus.off(Events.MANIFEST_UPDATED, update);
+        eventBus.off(Events.STREAM_TEARDOWN_COMPLETE, resetMetricsControllers);
```

Comment:
The call to attachSource (with a null parameter) in reset of MediaPlayer will trigger a reset call on StreamController. At the end of this function, the event STREAM_TEARDOWN_COMPLETE is sent....resetMetricsControllers is called at this time. According to my understanding, it is sufficient.